api/app/controllers/SlotsAdminController.scala

```diff
@@ -66,16 +69,39 @@ class SlotsAdminController(
 
   def update(slotId: String) = Action.async(parse.json[SlotUpdateParams]) {
     implicit request =>
-      val resp = store.getSlot(slotId)
-
-      resp map { slot =>
-        slot match {
-          case Some(slot) =>
-            val updatedSlot = SlotUpdateParams.toSlot(request.body, slot)
-            store.updateSlot(slotId, updatedSlot)
-            Ok(Json.toJson(Map("slot" -> updatedSlot)))
-          case None => NotFound
-        }
+      val validator = SlotValidator(store)
+
+      val result = for {
+        slotOpt <- store.getSlot(slotId)
+        slot <- toFut(slotOpt, new NotFoundException("Not found"))
+        updatedSlot = SlotUpdateParams.toSlot(request.body, slot)
+        validation <- handleValidationErrors(validator.validate(updatedSlot))
```

Comment:
Yep that makes sense. This seemed like a good fit because of the dependency on the `store` (which I can imagine being common to many validations we might do). But passing the store as a parameter to a function works just as well.